### PR TITLE
Commerce 1539 (5/7) Fix error handlings and casting

### PR DESCRIPTION
**Pull request by alecsloan**

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
@marco-leo Rebased

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:stop

**Comment by alecsloan** [comment]
ci:test

File: commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceOrderGenerator.java [hunk 1]
```diff
@@ -451,13 +452,13 @@ private int _randomInt(int min, int max) {
...
 
-		long range = max + 1 - min;
+		int range = max + 1 - min;
 
 		if (range == 0) {
 			return value;
 		}
 
-		return (int)(Math.abs(value) % range + min);
+		return value % range + min;
```
Comment (ethanbustad) [inline commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceOrderGenerator.java]: we should add parens here to clarify what the order of operations is.

File: commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceShipmentGenerator.java [hunk 2]
```diff
@@ -282,13 +282,13 @@ private int _randomInt(int min, int max) {
...
 
-		long range = max + 1 - min;
+		int range = max + 1 - min;
 
 		if (range == 0) {
 			return value;
 		}
 
-		return (int)(Math.abs(value) % range + min);
+		return value % range + min;
```
Comment (ethanbustad) [inline commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceShipmentGenerator.java]: same here

File: commerce-payment-method-authorize-net/src/main/java/com/liferay/commerce/payment/method/authorize/net/internal/AuthorizeNetCommercePaymentMethod.java [hunk 3]
```diff
@@ -96,6 +96,12 @@ public CommercePaymentResult completePayment(
 			CommercePaymentRequest commercePaymentRequest)
 		throws Exception {
 
+		if (!(commercePaymentRequest instanceof
+				AuthorizeNetCommercePaymentRequest)) {
+
+			throw new ClassCastException();
```
Comment (ethanbustad) [inline commerce-payment-method-authorize-net/src/main/java/com/liferay/commerce/payment/method/authorize/net/internal/AuthorizeNetCommercePaymentMethod.java]: Does this buy us anything? Wouldn't the next statement down throw a `ClassCastException` without this diff?

EDIT: same for all the diffs like this below.

File: commerce-product-type-virtual-order-service/src/main/java/com/liferay/commerce/product/type/virtual/order/service/impl/CommerceVirtualOrderItemLocalServiceImpl.java [hunk 4]
```diff
@@ -248,8 +248,8 @@ public File getFile(long commerceVirtualOrderItemId) throws Exception {
 			commerceOrderItem.getNameCurrentValue() + StringPool.PERIOD +
 				extension);
 
-		if (file.exists()) {
-			file.delete();
+		if (file.exists() && !file.delete()) {
+			throw new Exception();
```
Comment (ethanbustad) [inline commerce-product-type-virtual-order-service/src/main/java/com/liferay/commerce/product/type/virtual/order/service/impl/CommerceVirtualOrderItemLocalServiceImpl.java]: It'd be nice to throw a more specific exception here... maybe even just `IOException`?

**Review by ethanbustad — CHANGES_REQUESTED** [review]

File: commerce-payment-method-authorize-net/src/main/java/com/liferay/commerce/payment/method/authorize/net/internal/AuthorizeNetCommercePaymentMethod.java (hunk 3 above)
Comment (alecsloan) [reply commerce-payment-method-authorize-net/src/main/java/com/liferay/commerce/payment/method/authorize/net/internal/AuthorizeNetCommercePaymentMethod.java]: Yeah I'd imagine the logic before was fine. This is just the change that the javadoc requested

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Review by ethanbustad — APPROVED** [review]

**Pull request merged**